### PR TITLE
Add images for Ubuntu 24.04

### PR DESCRIPTION
**Pull request by richcarl**

### CODE REVIEW

**Comment by richcarl** [comment]
Why do these images build and install a standalone rocksdb library, separately from the one that is included in erlang_rocksdb? Are there versioning guarantees that need to be maintained between these? Upstream RocksDB is at version 9 something, and the image currently uses 7.10.2, while the code in erlang_rocksdb is a patched 7.7.3.

**Comment by hanssv** [comment]
I see this: https://github.com/aeternity/mnesia_rocksdb/commit/296813ef7b83bb4cddda19ac5df1759a98e77875 mentioning 7.10.2 so that is probably deliberate 🤔

**Comment by dincho** [comment]
> Why do these images build and install a standalone rocksdb library, separately from the one that is included in erlang_rocksdb? Are there versioning guarantees that need to be maintained between these? Upstream RocksDB is at version 9 something, and the image currently uses 7.10.2, while the code in erlang_rocksdb is a patched 7.7.3.

Probably `erlang_rocksdb` didn't included it back then. On the version numbers IDK, I guess application compatibility nobody used to up?

**Comment by richcarl** [comment]
> Probably `erlang_rocksdb` didn't included it back then. On the version numbers IDK, I guess application compatibility nobody used to up?

That's what I suspect. Curious if someone can confirm (and if we can drop this downloading and building of standalone rocksdb as part of the image).

**Comment by dincho** [comment]
> > Probably `erlang_rocksdb` didn't included it back then. On the version numbers IDK, I guess application compatibility nobody used to up?
> 
> That's what I suspect. Curious if someone can confirm (and if we can drop this downloading and building of standalone rocksdb as part of the image).

I wonder does the library supports multi-platforms (we currently only need arm64 and amd64)

P.S. I didn't approved your PR as the tests of the new builds are failing

**Comment by richcarl** [comment]
It was originally added by Ulf in #3376. Upstream did by default link statically with RocksDB but did not compile it with LZ4 enabled, which caused our tests to fail. There's a statement in the Dockerfile for Aeternity: ENV ERLANG_ROCKSDB_OPTS "-DWITH_SYSTEM_ROCKSDB=ON -DWITH_LZ4=ON ..." that sets up this. But when building locally, it's not enabled.

File: Dockerfile-noble [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+RUN OTP_DOWNLOAD_URL="https://github.com/erlang/otp/archive/OTP-${OTP_VERSION}.tar.gz" \
+    && curl -fsSL -o otp-src.tar.gz "$OTP_DOWNLOAD_URL" \
+    && mkdir otp-src \
+    && tar -zxf otp-src.tar.gz -C otp-src --strip-components=1 \
+    && cd otp-src \
+    && export ERL_TOP=`pwd` \
+    && ./otp_build autoconf && ./configure && make -j$(nproc) && make install
+
+ENV PORTABLE=1
+ENV ROCKSDB_VERSION=8.11.4
```
Comment (dincho) [inline Dockerfile-noble:42]: This version is different/newer than the other images, I'm a bit concerned about this diversion

**Comment by dincho** [comment]
> test Aeternity on Ubuntu 24 in CI.

I've no idea what the plan about this is, but it should not go into the the CI for sure

**Pull request merged**